IM_calculation/IM/im_calculation.py

```diff
@@ -94,6 +94,19 @@ def check_rotd(comps_to_store: Iterable[Components]) -> bool:
         )
     )
 
+def get_rotd_components_dict(rotd, comps_to_store: List[Components]):
+    value_dict = {}
+
+    rotd50 = np.median(rotd, axis=-1)
+    rotd100 = np.max(rotd, axis=-1)
+
+    if Components.crotd50 in comps_to_store:
+        value_dict[Components.crotd50.str_value] = rotd50
+    if Components.crotd100 in comps_to_store:
+        value_dict[Components.crotd100.str_value] = rotd100
+    if Components.crotd100_50 in comps_to_store:
+        value_dict[Components.crotd100_50.str_value] = rotd100 / rotd50
+    return value_dict
```

Comment:
This is out again

Reply:
Yes, it is intentional. This seperate function was needed for Vahid's calculation (which is urgent), which uses Tcl code for getting rotation. I just needed a way to return the rotd components. This branch is not going to be merged until I make it not depend on Tcl for rotation, which will make this separation unnecessary.

Reply:
Is this change still in progress?